pkg/config/expand.go

```diff
@@ -32,6 +32,7 @@ const (
 	blueprintLabel  string = "ghpc_blueprint"
 	deploymentLabel string = "ghpc_deployment"
 	roleLabel       string = "ghpc_role"
+	globalGroupID   string = "deployment"
```

Comment:
IIUC we never surface this "deployment" string to the user, at the same time it prevents (or even worse silently allows, but makes mistakes) users from naming they groups "deployment".
This change can be a good opportunity to solve this problem. As a simple solution can we replace this with less usable word e.g. "hpc_toolkit_internal_keyword_for_global_variables".

Reply:
I think this has some value but I'd rather not (potentially) protract this particular PR. There is a current task to reject deployment groups that are named "deployment" and using a reserved keyword for this value may be a better way to address that.